server/service/service.go

```diff
@@ -269,29 +288,165 @@ func (s *Service) ReportStatus(ctx context.Context, req *bpb.ReportStatusRequest
 	return &bpb.EmptyResponse{}, s.em.SetStatus(ctx, req)
 }
 
+const (
+	stateInitial = iota
+	// TPM 1.2 states
+	stateTPM12ChallengeSent
+	stateTPM12EKIdentitySent
+	// TPM 2.0 states
+	stateTPM20CSRRequested
+	stateTPM20NonceSent
+	// Common state
+	stateAttested
+)
+
+// Function to check if a client cert was presented
+func hasClientCert(ctx context.Context) bool {
```

Comment:
I think it's fine to remove this check. Reason is the Bootz server performs this validation already (e.g. the interceptor logic you wrote).